sql/core/src/test/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListenerSuite.scala

```diff
@@ -142,286 +163,277 @@ class SQLListenerSuite extends SparkFunSuite with SharedSQLContext with JsonTest
+    listener.onJobStart(SparkListenerJobStart(
       jobId = 0,
       time = System.currentTimeMillis(),
       stageInfos = Seq.empty,
       createProperties(executionId)))
-    bus.postToAll(SparkListenerSQLExecutionEnd(
+    listener.onOtherEvent(SparkListenerSQLExecutionEnd(
       executionId, System.currentTimeMillis()))
-    bus.postToAll(SparkListenerJobEnd(
+    listener.onJobEnd(SparkListenerJobEnd(
       jobId = 0,
       time = System.currentTimeMillis(),
       JobFailed(new RuntimeException("Oops"))
     ))
 
-    assertJobs(store.execution(0), failed = Seq(0))
+    assertJobs(statusStore.execution(executionId), failed = Seq(0))
   }
 
   test("SPARK-11126: no memory leak when running non SQL jobs") {
-    val previousStageNumber = statusStore.executionsList().size
-    spark.sparkContext.parallelize(1 to 10).foreach(i => ())
-    spark.sparkContext.listenerBus.waitUntilEmpty(10000)
-    // listener should ignore the non SQL stage
-    assert(statusStore.executionsList().size == previousStageNumber)
-
-    spark.sparkContext.parallelize(1 to 10).toDF().foreach(i => ())
-    spark.sparkContext.listenerBus.waitUntilEmpty(10000)
-    // listener should save the SQL stage
-    assert(statusStore.executionsList().size == previousStageNumber + 1)
-  }
-
-  test("driver side SQL metrics") {
-    val oldCount = statusStore.executionsList().size
-    val expectedAccumValue = 12345L
-    val physicalPlan = MyPlan(sqlContext.sparkContext, expectedAccumValue)
-    val dummyQueryExecution = new QueryExecution(spark, LocalRelation()) {
-      override lazy val sparkPlan = physicalPlan
-      override lazy val executedPlan = physicalPlan
-    }
-
-    SQLExecution.withNewExecutionId(spark, dummyQueryExecution) {
-      physicalPlan.execute().collect()
-    }
-
-    while (statusStore.executionsList().size < oldCount) {
-      Thread.sleep(100)
-    }
-
-    // Wait for listener to finish computing the metrics for the execution.
-    while (statusStore.executionsList().last.metricValues == null) {
-      Thread.sleep(100)
+    val statusStore = createStatusStore()
+    val listener = statusStore.listener.get
+    try {
+      sparkContext.addSparkListener(listener)
+      spark.sparkContext.parallelize(1 to 10).foreach(i => ())
+      spark.sparkContext.listenerBus.waitUntilEmpty(10000)
+      // Listener should ignore the non-SQL stages, as the stage data are only removed when SQL
+      // execution ends, which will not be triggered for non-SQL jobs.
+      assert(listener.stageMetrics.isEmpty)
+    } finally {
+      sparkContext.removeSparkListener(listener)
     }
-
-    val execId = statusStore.executionsList().last.executionId
-    val metrics = statusStore.executionMetrics(execId)
-    val driverMetric = physicalPlan.metrics("dummy")
-    val expectedValue = SQLMetrics.stringValue(driverMetric.metricType, Seq(expectedAccumValue))
-
-    assert(metrics.contains(driverMetric.id))
-    assert(metrics(driverMetric.id) === expectedValue)
```

Comment:
ditto, but I don't know why this passed before...

Reply:
This passed before because the listener was automatically being added to the bus using the plugin interface you've removed in this PR.

Reply:
In fact this will probably still pass if you restore the `LIVE_ENTITY_UPDATE_PERIOD` config in the session, since you'll still have a listener in the shared session (it's should still be installed automatically by your code, just not using the plugin).